internal/uvm/scsi.go

```diff
@@ -55,40 +100,107 @@ func (uvm *UtilityVM) allocateSCSI(ctx context.Context, hostPath string, uvmPath
 func (uvm *UtilityVM) deallocateSCSI(ctx context.Context, controller int, lun int32) {
 	uvm.m.Lock()
 	defer uvm.m.Unlock()
-	si := uvm.scsiLocations[controller][lun]
-	if si.hostPath != "" {
+	if si := uvm.scsiLocations[controller][lun]; si != nil {
 		log.G(ctx).WithFields(logrus.Fields{
-			"hostPath":   si.hostPath,
-			"uvmPath":    si.uvmPath,
+			"hostPath":   si.HostPath,
+			"uvmPath":    si.UvmPath,
 			"isLayer":    si.isLayer,
 			"refCount":   si.refCount,
 			"controller": controller,
 			"lun":        lun,
 		}).Debug("removed SCSI location")
-		uvm.scsiLocations[controller][lun] = scsiInfo{}
+		uvm.scsiLocations[controller][lun] = nil
 	}
 }
 
 // Lock must be held when calling this function.
 func (uvm *UtilityVM) findSCSIAttachment(ctx context.Context, findThisHostPath string) (int, int32, string, error) {
```

Comment:
Can this just return si instead of various fields from si?

Reply:
Yep, this was just changed